src/storage/protocols/tus/s3-locker.ts

```diff
@@ -120,7 +120,7 @@ export class S3Locker implements Locker {
     return false
   }
 
-  async renewLock(id: string): Promise<boolean> {
+  async renewLock(id: string, checkLocked: () => boolean): Promise<boolean> {
```

Comment:
Any specific reason why we need to pass a function a not check directly this.isLocked?

Reply:
This is in `renewLock()` wich is a part of `S3Locker`, but `isLocked` is part of `S3Lock`. So when S3Lock calls `this.locker.renewLock()` it passes in a function to allow S3Locker to check the lock state of S3Lock.

Alternatives would be
- move the lock states to S3Locker with a `map<string, boolean>` (id -> isLocked) and share it that way
- pass in the whole S3Lock instance (instead of checkLocked) and expose isLocked from S3Lock (currently private) so the locker can check the lock directly